sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RedirectPolicy.java

```diff
@@ -62,7 +62,13 @@ public RedirectPolicy(RedirectStrategy redirectStrategy) {
                 if (redirectStrategy.shouldAttemptRedirect(context, httpResponse, redirectAttempt,
                     attemptedRedirectUrls)) {
                     HttpRequest redirectRequestCopy = redirectStrategy.createRedirectRequest(httpResponse);
-                    return httpResponse.getBody()
+
+                    // Clear the authorization header to avoid the client to be redirected to an untrusted third party server
+                    // causing it to leak your authorization token to.
+                    httpResponse.getHeaders().remove("Authorization");
```

Comment:
If we remove the "Authorization" header, how does the redirected request get authorized?

Reply:
I think it is expected behavior to remove the authorization header before we redirect for security reasons.
So that we don't forward the authorization token to an untrusted/unwanted site.

It could be the service's responsibility to set the auth tokens accordingly in the redirect request so that it is forwarded correctly I would think.

Ref: https://github.com/Azure/azure-sdk-for-net/pull/22876#discussion_r691688302